omf-bifrost/overture-valhalla-writer/src/writer.rs

```diff
@@ -269,46 +324,100 @@ fn process_segment(
+
+    // TODO: Can we leave this 0 for Overture->Valhalla conversion?
+    way.set_has_user_tags_(0);
+
+    // TODO: Have a second look at this, does this mean pedestrian-only?
+    way.set_pedestrian_forward_(1);
+    way.set_pedestrian_backward_(1);
+
+    // TODO: get this from Overture data
+    way.speed_ = 25; // 25 km/h
+
+    way
+}
+
+fn create_valhalla_osmwaynode(way_index : u32, way_shape_node_index : u32, osmid: u64, lng: f64, lat: f64, intersection: u32) -> OSMWayNode
+{
+    let mut waynode = OSMWayNode::default();
+    waynode.way_index = way_index;
+    waynode.way_shape_node_index = way_shape_node_index;
+
+    waynode.node.osmid_ = osmid;
+
+    let (lat7, lng7) = utils::encode_lat_lon(lat, lng);
+    waynode.node.lng7_ = lng7;
+    waynode.node.lat7_ = lat7;
+    waynode.node.set_intersection_(intersection);
+
+    // TODO: could also be 4095 ("kAllAccess")? See "graphconstants.h" in Valhalla
+    // TODO: get from Overture data
+    waynode.node.set_access_(2047);
+
+    waynode
+}
+
+fn export_roads(exported_roads: &Vec<ExportedRoad>, output_dir: &str)
+{
+    let mut ways = Vec::new();
+    let mut waynodes = Vec::new();
+
+    for way_index in 0..exported_roads.len() {
+        let node_count = exported_roads[way_index].points.len() as u16;
+        let offset_way_index: u64 = way_index as u64 * 2;
+        ways.push(create_valhalla_osmway(offset_way_index + 1, 1, node_count));
+        ways.push(create_valhalla_osmway(offset_way_index + 2, 1, node_count));
+
+        // Valhalla complains when road is only one way, so for now we export it twice, this is the first time...
+        for (point_index, point) in exported_roads[way_index].points.iter().enumerate() {
+            // TODO: only make intersection if other way intersects
+            let intersection: u64 = 1;
+
+            waynodes.push(create_valhalla_osmwaynode(
+                offset_way_index as u32,
+                point_index as u32,
+                point.index as u64,
+                point.point.lon, point.point.lat,
+                intersection as u32
+            ));
         }
 
-        osm::write_osmways_to_file(ways, (exported_roads.len() * 2) as u64, &format!("{}/ways.bin", output_dir));
-        osm::write_osmwaynodes_to_file(waynodes, total_way_nodes, &format!("{}/way_nodes.bin", output_dir));
+        // ... and this is the second time.
+        for (point_index, point) in exported_roads[way_index].points.iter().rev().enumerate() {
+            // TODO: only make intersection if other way intersects
+            let intersection: u64 = 1;
+
+            waynodes.push(create_valhalla_osmwaynode(
+                offset_way_index as u32,
+                point_index as u32,
+                point.index as u64,
+                point.point.lon, point.point.lat,
+                intersection as u32
+            ));
+        }
     }
+
+    ways.write_to_file(&format!("{}/ways.bin", output_dir));
```

Comment:
don't manipulate paths by hand - won't work on windows. Make sure `output_dir` is a Path or PathBuf, and use [join](https://doc.rust-lang.org/std/path/struct.Path.html#method.join):  `output_dir.join("ways.bin")`